packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -35,14 +35,20 @@ export default function defaultResolver(
 
   const resolve = options.browser ? browserResolve : resolveSync;
 
-  return resolve(path, {
+  let result = resolve(path, {
     basedir: options.basedir,
     defaultResolver,
     extensions: options.extensions,
     moduleDirectory: options.moduleDirectory,
     paths: options.paths,
     rootDir: options.rootDir,
   });
+  if (options.browser && result) {
+    // Dereference symlinks to ensure we don't create a separate
+    // module instance depending on how it was referenced.
+    result = fs.realpathSync(result);
```

Comment:
should use `realpath-native`

Reply:
Should this usage be changed also https://github.com/facebook/jest/blob/964dc1619b8c5345a648ef79bcd87c087995255a/packages/jest-resolve/src/defaultResolver.ts#L104?

Reply:
yeah, let's